### DISCUSSION TITLE
What are the target platforms and operating systems for thorvg?

### DISCUSSION THREAD

**Discussion by xlbao123** [thorvg/thorvg#1783] [Q&A] [answered]
I've seen that thorvg requires C++14 and utilizes C++ multithreading interfaces. Does this mean that it is limited to platforms like Linux, Windows, or macOS? The rendering now supports opengl and software rendering (swrender). So, to smoothly run most SVG/Lottie files, is it a must to have a 3D GPU? If using swrender, are there any specific hardware requirements, such as a processor greater than 1GHz Cortex-A? 
I want to run thorvg smoothly on a microcontroller, typically with 300MHZ Cortex-M, a simple 2D GPU (supporting basic point and line drawing operations), and running an RTOS. I'm unsure if such a platform is within thorvg's target considerations.

**Comment by hermet** [comment]
Hi, essentially, ThorVG has no dependencies, guarantee POSIX standard. It can be run with an RTOS if your RTOS toolchain supports C++14 or higher.  Actual performance may vary depending on factors such as program size, screen resolution, multi-processing capacity, and other runtime environmental considerations. Currently, we only guarantee the software renderer for ThorVG, and OpenGL usage is not very well. Despite this, the software renderer enables faster drawing with a decent rasterizer and low memory usage. It is likely to perform well on modern embedded CPUs, including 1GHz Cortex-A you mentioned(I guess so), unless the screen resolution is too high. We have successfully applied a prior version engine of ThorVG to the Samsung Galaxy Watch devices years ago with a resolution of 360 x 360 and a Cortex-A processor running at 1.6GHz.

**Comment by hermet** [comment]
And I'm not certain about the performance with a "300MHz Cortex-M" since I haven't tested ThorVG with it before. However, I assume it should work fine, particularly if the graphics components are relatively static (i.e., not targeting 60fps animation).

**Comment by xlbao123** [comment]
> And I'm not certain about the performance with a "300MHz Cortex-M" since I haven't tested ThorVG with it before. However, I assume it should work fine, particularly if the graphics components are relatively static (i.e., not targeting 60fps animation).

Sure, I will try to port it and provide feedback on any issues.

**Comment by hermet** [comment, marked as answer]
@xlbao123 mark as an answer?
